module/VuFind/src/VuFind/Db/Service/RecordService.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * Database service for Records.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+
+namespace VuFind\Db\Service;
+
+use Exception;
+use VuFind\Db\Entity\RecordEntityInterface;
+use VuFind\Db\Table\DbTableAwareInterface;
+use VuFind\Db\Table\DbTableAwareTrait;
+
+/**
+ * Database service for Records.
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+class RecordService extends AbstractDbService implements DbTableAwareInterface, RecordServiceInterface
+{
+    use DbTableAwareTrait;
+
+    /**
+     * Find a record by id.
+     *
+     * @param string $id     Record ID
+     * @param string $source Record source
+     *
+     * @return ?RecordEntityInterface
+     */
+    public function findRecord($id, $source): ?RecordEntityInterface
```

Comment:
This is a nitpick, but most services have methods that start with 'get' instead of 'find' (like `getUserById`). Is there a reason not to use `getRecord` here and `getRecords` below? I know the table class has `findRecord`, but it bugs me..

Reply:
Yes, good idea. I was being lazy because renaming will make the merge conflicts with #2233 worse, but it's worth the effort!